File: 001/1.java
// Find the sum of all the multiples of 3 or 5 below 1000.

class p1 {

  static int sumUnder(int limit) {
    return ((limit * limit) + limit) / 2;
  }

  static int sumSteps(int step, int limit) {
    return step * sumUnder((limit - 1) / step);
  }

  public static void main(String[] args) {
    System.out.println(sumSteps(3, 1000) + sumSteps(5, 1000) - sumSteps(15, 1000));
  }
}
// 233168

File: 001/1.zig
// The natural numbers below 10 that are multiples of 3 or 5 are 3, 5, 6 and 9.
// The sum of these multiples is 23.
// Find the sum of all the multiples of 3 or 5 below 1000.

const std = @import("std");
const stdout = std.io.getStdOut().writer();

fn sumUnder(x: u32) u32 {
    return ((x * x) + x) / 2;
}

fn sumSteps(step: u32, limit: u32) u32 {
    return sumUnder((limit - 1) / step) * step;
}

pub fn main() !void {
    const x = sumSteps(3, 1000) + sumSteps(5, 1000) - sumSteps(15, 1000);
    try stdout.print("{d}\n", .{x});
}
